__tests__/integration/hathorwallet_others.test.js

```diff
@@ -0,0 +1,1247 @@
+
+describe('getAddressInfo', () => {
+  /** @type HathorWallet */
+  let hWallet;
+  beforeAll(async () => {
+    hWallet = await generateWalletHelper();
+  });
+
+  afterAll(() => {
+    hWallet.stop();
+  });
+
+  it('should display correct values for HTR transactions with no change', async () => {
+    const addr0 = hWallet.getAddressAtIndex(0);
+    const addr1 = hWallet.getAddressAtIndex(1);
+
+    // Validating empty address information
+    expect(hWallet.getAddressInfo(addr0)).toMatchObject({
+      total_amount_received: 0,
+      total_amount_sent: 0,
+      total_amount_available: 0,
+      total_amount_locked: 0, // Validating this field only once to check it's returned
+      token: HATHOR_TOKEN_CONFIG.uid, // Validating this field only once to ensure it's correct
+      index: 0,
+    });
+
+    // Validating address after 1 transaction
+    await GenesisWalletHelper.injectFunds(addr0, 10);
+    expect(hWallet.getAddressInfo(addr0)).toMatchObject({
+      total_amount_received: 10,
+      total_amount_sent: 0,
+      total_amount_available: 10,
+    });
+
+    // Validating the results for two transactions
+    let tx = await hWallet.sendTransaction(addr1, 10);
+    await waitForTxReceived(hWallet, tx.hash);
+    expect(hWallet.getAddressInfo(addr0)).toMatchObject({
+      total_amount_received: 10,
+      total_amount_sent: 10,
+      total_amount_available: 0,
+      index: 0, // Ensuring the index is correct
+    });
+    expect(hWallet.getAddressInfo(addr1)).toMatchObject({
+      total_amount_received: 10,
+      total_amount_sent: 0,
+      total_amount_available: 10,
+      index: 1, // Ensuring the index is correct
+    });
+
+    // Validating the results for the funds returning to previously used address
+    await waitUntilNextTimestamp(hWallet, tx.hash);
+    tx = await hWallet.sendTransaction(addr0, 10);
+    await waitForTxReceived(hWallet, tx.hash);
+    expect(hWallet.getAddressInfo(addr0)).toMatchObject({
+      total_amount_received: 20,
+      total_amount_sent: 10,
+      total_amount_available: 10,
+    });
+    expect(hWallet.getAddressInfo(addr1)).toMatchObject({
+      total_amount_received: 10,
+      total_amount_sent: 10,
+      total_amount_available: 0,
+    });
+  });
+
+  it('should throw for an address outside the wallet', async () => {
+    expect(() => hWallet.getAddressInfo(WALLET_CONSTANTS.genesis.addresses[0])).toThrow();
+  });
+
+  it('should display correct values for transactions with change', async () => {
+    const addr2 = hWallet.getAddressAtIndex(2);
+    const addr3 = hWallet.getAddressAtIndex(3);
+
+    // Ensure both are empty addresses
+    expect(hWallet.getAddressInfo(addr2).total_amount_received).toStrictEqual(0);
+    expect(hWallet.getAddressInfo(addr3).total_amount_received).toStrictEqual(0);
+
+    // Move all the wallet's funds to "2"
+    let tx = await hWallet.sendTransaction(addr2, 10);
```

Comment:
Since we know the tests inside a `describe` block will be executed linearly, I declared the `hWallet` HathorWallet on the `describe` block scope. That way and all the `it` blocks reuse the same wallet and inherit the state from the last test.

This was done because the test's scope was restricted to one address at a time, which seemed to allow for a shared wallet. Do you think this made it too confusing?

Reply:
I though the wallet was being started with a `beforeEach` which would change the wallet every test.